test/tests/functional/pbs_cray_vnode_pool.py

```diff
@@ -0,0 +1,337 @@
+                                   starttime=start_time)
+        self.assertTrue(rc)
+
+        rc = self.momA.log_match(
+            "Hello (no inventory required) from server", max_attempts=9,
+            starttime=start_time)
+        if rc is None:
+            found_in_momA = 0
+        else:
+            found_in_momA = 1
+
+        rc = self.momB.log_match(
+            "Hello (no inventory required) from server", max_attempts=9,
+            starttime=start_time)
+        if rc is None:
+            found_in_momB = 0
+        else:
+            found_in_momB = 1
+
+        self.assertEqual(found_in_momA + found_in_momB,
+                         1, msg="an inventory mom not chosen correctly")
+
+        # Only one mom is inventory mom
+        if (found_in_momA == 0):
+            self.inv_mom = self.momA
+            self.noninv_mom = self.momB
+        else:
+            self.inv_mom = self.momB
+            self.noninv_mom = self.momA
+
+        self.logger.info("Inventory mom is %s." % self.inv_mom.shortname)
+        self.logger.info("Non-inventory mom is %s." %
+                         self.noninv_mom.shortname)
+
+        # Kill inventory mom
+        self.inv_mom.stop(sig='-KILL')
+
+        # Check that former inventory mom is down
+        rv = self.server.expect(
+            VNODE, {'state': 'down'}, id=self.inv_mom.shortname,
+            max_attempts=10, interval=2)
+        self.assertTrue(rv)
+
+        # Check if inventory mom changed and is listed in the server log.
+        rc = self.server.log_match(
+            "Setting inventory_mom for vnode_pool %s to %s" %
+            ('1', self.noninv_mom.shortname), max_attempts=5,
+            starttime=start_time)
+        if rc is None:
+            self.logger.info(
+                "Inventory mom %s is not in server logs." %
+                (self.noninv_mom.shortname))
+            self.assertTrue(rc)
+        else:
+            self.logger.info(
+                "Inventory mom is now %s in server logs." %
+                (self.noninv_mom.shortname))
+            self.assertTrue(rc)
+
+        # Check compute nodes are up
+        vlist = []
+        try:
+            vnl = self.server.filter(
+                VNODE, {'resources_available.vntype': 'cray_compute'})
+            vlist = vnl["resources_available.vntype=cray_compute"]
+        except Exception:
+            sys.exc_clear()
+
+        # Loop through each compute vnode in the list and check if state = free
+        r1 = len(vlist)
+        for i in range(0, r1):
+            v1 = vlist[i]
+            # Check that the node is in free state
+            rv = self.server.expect(
+                VNODE, {'state': 'free'}, id=v1, max_attempts=3, interval=2)
+            self.assertTrue(rv)
+
+        start_time1 = int(time.time())
+        # Start the previous inv mom
+        self.inv_mom.start()
+
+        # Check previous inventory mom is up and no longer the inventory mom
```

Comment:
The EDD mentions "If the Inventory Mom goes down, then another Mom from the same vnode_pool is chosen to be the Inventory Mom." but does not say that once the previous inventory mom is back up that it will no longer be the inventory mom.  The doc string and the comment in line 216 needs updating too.  Starting the previous inventory mom might be useful only for the purpose of having two moms up which might be needed by the other tests.